src/Nethermind/Nethermind.Consensus/Producers/TxPoolTxSource.cs

```diff
@@ -126,18 +123,19 @@ private void SelectBlobTransactions(IEnumerable<Transaction> blobTransactions, B
             UInt256 blobGasCounter = 0;
             UInt256 feePerBlobGas = UInt256.Zero;
 
+            var maxBlobGasPerBlock = spec.MaxBlobCount * Eip4844Constants.GasPerBlob;
             foreach (Transaction blobTx in blobTransactions)
             {
-                if (blobGasCounter >= _eip4844Config.MaxBlobGasPerBlock)
+                if (blobGasCounter >= maxBlobGasPerBlock)
                 {
                     if (_logger.IsTrace) _logger.Trace($"Declining {blobTx.ToShortString()}, no more blob space. Block already have {blobGasCounter} blob gas which is max value allowed.");
                     break;
                 }
 
                 checkedBlobTransactions++;
 
-                ulong txBlobGas = (ulong)(blobTx.BlobVersionedHashes?.Length ?? 0) * _eip4844Config.GasPerBlob;
-                if (txBlobGas > _eip4844Config.MaxBlobGasPerBlock - blobGasCounter)
+                ulong txBlobGas = (ulong)(blobTx.BlobVersionedHashes?.Length ?? 0) * Eip4844Constants.GasPerBlob;
```

Comment:
refactor?

Reply:
I think it's fine, gas per blob remain the same